project_template/inc/az/exampleshortname/operations.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/** @file operations.h */
+#pragma once
+#include <stdint.h>
+#include <az/exampleshortname/export.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+/** @brief add two numbers together
+ *  @param[in] a
+ *  @param[in] b
+ *  @return @p a + @p b

```

Comment:
>@p [](start = 12, length = 2)

Similar to the parameters, consider putting the returned type here. Someone reading the documentation will want to know what the type is of the return value.

Reply:
isn't that already covered by the declaration of the function? The return type will be included in the docs automatically.

I think doxygen may actually even include the `int` return type for functions declared in K&R style (default int), but we won't be allowing that.

Reply:
it does not. it will put the function with the input/output, function declaration etc in the top level but you need to manually write the documentation for the different parts of it. you can see how we do it [here](https://github.com/Azure/azure-ulib-c/blob/master/inc/ustream.h) with [these](https://azure.github.io/azure-ulib-c/ustream_8h.html) results.

Reply:
OK I like using @retval for describing functions with special return values. But here there are no special return values.

Reply:
understood and agree. but i still think the standard we set should be that we lay out explicitly the type of the return. so here it might be `The @int32_t with the result of the addition` or something like that.